### PR TITLE
Add PEP 621 tab for entry points docs

### PR DESCRIPTION
**Pull request by flying-sheep**
Since #1031 is merged, let’s gradually move away from a setuptools centric view.

### [Rendered](https://python-packaging-user-guide--1115.org.readthedocs.build/en/1115/guides/creating-and-discovering-plugins/#using-package-metadata)

### CODE REVIEW

File: source/guides/creating-and-discovering-plugins.rst [hunk 1]
```diff
@@ -120,21 +120,29 @@ a list of packages to :func:`setup`'s ``packages`` argument instead of using
...
+Packages have metadata for plugins described in the :doc:`entry-points`.
+By telling your package’s build backend,
+plugins can register themselves for discovery.
 
 For example if you have a package named ``myapp-plugin-a`` and it includes
-in its :file:`setup.py`:
+the following in its setup file:
 
-.. code-block:: python
+.. tab:: :file:`pyproject.toml` (PEP 621)
```
Comment (bhrutledge) [inline source/guides/creating-and-discovering-plugins.rst:130]: I don't think we should reference `setup.py` at all, and just show `pyproject.toml`, without a reference to PEP 621.

File: source/guides/creating-and-discovering-plugins.rst [hunk 2]
```diff
@@ -120,21 +120,29 @@ a list of packages to :func:`setup`'s ``packages`` argument instead of using
 Using package metadata
 ======================
 
-:doc:`Setuptools <setuptools:index>` provides :doc:`special support
-<setuptools:userguide/entry_point>` for plugins. By providing the
-``entry_points`` argument to :func:`setup` in :file:`setup.py` plugins can
-register themselves for discovery.
+Packages have metadata for plugins described in the :doc:`entry-points`.
+By telling your package’s build backend,
+plugins can register themselves for discovery.
```
Comment (bhrutledge) [inline source/guides/creating-and-discovering-plugins.rst:125]: I think the language could be clearer here. Specifically, "telling your package's build backend" seems vague; what does "telling" mean, and do we need to mention "build backend" at all?

Also, it looks like this intends to link to the [entry points spec](https://packaging.python.org/en/latest/specifications/entry-points/), but it's not in the [PR preview](https://python-packaging-user-guide--1115.org.readthedocs.build/en/1115/guides/creating-and-discovering-plugins/). Perhaps it needs to be a `:ref:` instead?

**Review by bhrutledge — CHANGES_REQUESTED** [review]

File: source/guides/creating-and-discovering-plugins.rst (hunk 2 above)
Comment (flying-sheep) [reply source/guides/creating-and-discovering-plugins.rst]: both done!

**Review by Harlee1994 — APPROVED** [review]

**Comment by flying-sheep** [comment]
Hi @bhrutledge, can this be merged?

**Review by bhrutledge — APPROVED** [review]
Thanks for the nudge!

**Pull request merged**